sci-libs/linux-gpib/linux-gpib-4.3.6-r100.ebuild

```
@@ -0,0 +1,198 @@
+HOMEPAGE="https://linux-gpib.sourceforge.io/"
+SRC_URI="https://downloads.sourceforge.net/linux-gpib/${P}.tar.gz
+	firmware? ( https://linux-gpib.sourceforge.io/firmware/gpib_firmware-${FW_PV}.tar.gz )
+"
+S="${WORKDIR}/${PN}-user-${PV}"
+
+LICENSE="GPL-2"
+SLOT="0"
+KEYWORDS="~amd64 ~arm ~x86"
+IUSE="pcmcia static guile perl php python tcl doc firmware"
+REQUIRED_USE="
+	guile? ( ${GUILE_REQUIRED_USE} )
+	python? ( ${PYTHON_REQUIRED_USE} )
+"
+
+COMMONDEPEND="
+	sys-libs/readline:=
+	tcl? ( dev-lang/tcl:0= )
+	guile? ( ${GUILE_DEPS} )
+	perl? ( dev-lang/perl:= )
+	php? ( dev-lang/php:= )
+	python? ( ${PYTHON_DEPS} )
+	firmware? ( sys-apps/fxload )
+"
+RDEPEND="${COMMONDEPEND}
+	acct-group/gpib
+	~sci-libs/linux-gpib-modules-${PV}
+"
+DEPEND="${COMMONDEPEND}"
+BDEPEND="
+	virtual/pkgconfig
+	doc? ( app-text/docbook-sgml-utils )
+	perl? ( virtual/perl-ExtUtils-MakeMaker )
+"
+
+PATCHES=(
+	"${FILESDIR}"/${PN}-4.3.0-perl.patch
+)
+
+pkg_setup() {
+	use guile && guile-single_pkg_setup
+	use perl && perl_set_version
+	use python && python_setup
+}
+
+src_unpack() {
+	default
+	unpack "${WORKDIR}/${P}/${PN}-user-${PV}.tar.gz"
+}
+
+src_prepare() {
+	default
+
+	use guile && guile_bump_sources
+
+	# We have to use --root instead of --prefix for setup.py.
+	# Otherwise the python files are not installed properly into site-packages.
+	sed -i -e \
+		's/--prefix=$(DESTDIR)$(prefix)/--root=$(DESTDIR)/g' \
+		language/python/Makefile.am || die
+
+	eautoreconf
+}
+
+src_configure() {
+	econf \
+		$(use_enable static) \
+		$(use_enable guile guile-binding) \
+		$(use_enable perl perl-binding) \
+		$(use_enable php php-binding) \
+		$(use_enable python python-binding) \
+		$(use_enable tcl tcl-binding) \
+		$(use_enable doc documentation)
+}
+
+src_install() {
+	# PYTHONDONTWRITEBYTECODE=0 is required, otherwise
+	# installation of the python files is skipped
+	FIRM_DIR=/usr/share/usb
+	emake \
+		DESTDIR="${ED}" \
+		INSTALL_MOD_PATH="${ED}" \
+		HOTPLUG_USB_CONF_DIR=/etc/hotplug/usb \
+		UDEV_RULES_DIR="$(get_udevdir)"/rules.d \
+		USB_FIRMWARE_DIR=${FIRM_DIR} \
+		PYTHONDONTWRITEBYTECODE=0 \
+		docdir="/usr/share/doc/${PF}/html" install
+
+	use guile && guile_unstrip_ccache
+
+	if use perl; then
+		einfo "Installing perl module"
+		cd "${S}"/language/perl || die
+		emake DESTDIR="${ED}" install
+		perl_fix_packlist
+		perl_delete_emptybsdir
+		cd "${S}" || die
+	fi
+
+	echo "KERNEL==\"gpib[0-9]*\",	MODE=\"0660\", GROUP=\"gpib\"" >> 99-gpib.rules
+	udev_dorules 99-gpib.rules
+
+	dodoc doc/linux-gpib.pdf ChangeLog AUTHORS README* NEWS
+
+	insinto /etc
+	newins util/templates/gpib.conf gpib.conf
+	newins util/templates/gpib.conf gpib.conf.example
+
+	if use pcmcia; then
+		insinto /etc/pcmcia
+		doins "${S}"/etc/pcmcia/*
+	fi
+
+	# fix rules files
+	local f
+	find "${ED}$(get_udevdir)"/rules.d -type f -print0 | while read -rd '' f ; do
+		grep -q "${ED}" "${f}" && einfo "File ${f} contains a temporary path, fixing."
+		sed -i -e "s:${ED}:/:g" "${f}"
```

```suggestion
		sed -i -e "s:${ED}:/:g" "${f}" || die
```

but consider rewriting this as:
```
while read -rd '' f ; do
	....
done < <(...)
```

---

`find "${ED}$(get_udevdir)"/rules.d -type f -exec sed -i -e "s:${ED}:/:g" {} + || die`?

Although I actually dont see any need for this.

---

Looking at the source they appear to be using DESTDIR and UDEV_LIBDIR properly. ED shouldnt be leaking there. Maybe ten years ago it was different?